xnmt/input_reader.py

```diff
@@ -233,25 +244,6 @@ def count_words(self, trg_words):
   def vocab_size(self):
     return len(self.vocab)
 
-class CharFromWordTextReader(PlainTextReader, Serializable):
```

Comment:
I don't think this necessarily needs to be moved to the specialized place, it seems general enough, for example it could be used in any model that uses character-based representations of words. However, it does need to be documented and given type annotations. I don't think it should be a `SimpleSentence` but rather a `SegmentedSentence` or something.

I also think perhaps a better representation would be a list of lists, where each interior list represents a single word, but perhaps `SegmentedSentence` could provide options to convert between the two representations seamlessly.

Reply:
Sure! I would just move it back and add a TODO for @philip30 if that's ok?

Reply:
Sure.